server/utils/helpers.ts

```diff
@@ -81,3 +85,23 @@ export const prettifyErrorMessage = (rawErrorMessage: string) => {
     return `User ${match[2]} has no permissions to [${match[1]}].`;
   }
 };
+
+export function getClientBasedOnDataSource(
+  context: any,
+  dataSourceEnabled: boolean,
+  request: OpenSearchDashboardsRequest,
+  dataSourceId: string,
+  client: any
+): (
```

Comment:
from following the codepath, the client could be `undefined` here based on server/plugin.ts. Can we make this more typesafe by explicitly calling out `client` could be undefined here? Some comments when it is initialized would be helpful as well, and more details stating that if MDS is enabled, there will be no available client, since we fetch the configured client from the datasource plugin.

Reply:
Are you going to make this change?

Reply:
already made the change. And just realized I have already added comment in this function, so no need to add comments

Reply:
Can you please make the parameters more typesafe (context, client)? Again it is much easier to follow if we know what `client` is and that it could be undefined. Ideally having a comment about how it could be undefined is helpful. It's just for tracking and readability purposes for the future.

Reply:
I already did, 3 revisions ago. Please look at the latest code

Reply:
Ah! Missed that with all of the commit filtering. My bad. thanks!